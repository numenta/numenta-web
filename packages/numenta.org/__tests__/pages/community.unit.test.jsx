import React from 'react'
import renderer from 'react-test-renderer'
import stubContext from 'react-stub-context'

import context from '__tests__/__mocks__/reactContextMock'

import CommunityPage from '../../pages/community'


describe('Community Page React component', () => {

  it('Renders correctly', () => {
    const CommunityPageStubbed = stubContext(CommunityPage, context)
    const component = renderer.create(
      <CommunityPageStubbed />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
