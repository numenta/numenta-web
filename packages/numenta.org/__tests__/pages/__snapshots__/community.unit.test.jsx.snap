exports[`Community Page React component Renders correctly 1`] = `
<div>
  <div>
    <section
      className="section"
      id="sectionHome">
      <h1
        className="headline">
        Home of the HTM Community
      </h1>
      <div
        className="article">
        <article
          className="columns">
          <div
            className="aside">
            <aside
              className="glance">
              <div>
                <div
                  className="glanceHead">
                  Community Resources
                </div>
              </div>
              <dl
                className="defineList horizontal">
                <dt
                  className="defineTitle horizontal">
                  Discussion
                </dt>
                <dd
                  className="defineItem horizontal">
                  <a
                    className="textlink"
                    href="https://discourse.numenta.org/categories"
                    onClick={[Function]}
                    target={undefined}>
                    HTM Forum
                  </a>
                   
                   
                </dd>
                <dt
                  className="defineTitle horizontal">
                  Code
                </dt>
                <dd
                  className="defineItem horizontal">
                  <a
                    className="textlink"
                    href="https://github.com/numenta"
                    onClick={[Function]}
                    target={undefined}>
                    Numenta Github
                  </a>
                  <br />
                  <a
                    className="textlink"
                    href="https://github.com/htm-community"
                    onClick={[Function]}
                    target={undefined}>
                    HTM Community Github
                  </a>
                </dd>
                <dt
                  className="defineTitle horizontal">
                  Videos
                </dt>
                <dd
                  className="defineItem horizontal">
                  <a
                    className="textlink"
                    href="https://www.youtube.com/c/NumentaOrg"
                    onClick={[Function]}
                    target={undefined}>
                    YouTube
                  </a>
                </dd>
                <dt
                  className="defineTitle horizontal">
                  Technical
                </dt>
                <dd
                  className="defineItem horizontal">
                  <a
                    className="textlink"
                    href="http://numenta.com/papers/"
                    onClick={[Function]}
                    target={undefined}>
                    Papers
                  </a>
                </dd>
                <dt
                  className="defineTitle horizontal">
                  License
                </dt>
                <dd
                  className="defineItem horizontal">
                  <a
                    className="textlink"
                    href={undefined}
                    onClick={[Function]}
                    style={Object {}}
                    target={undefined}>
                    AGPLv3
                  </a>
                </dd>
                <dt
                  className="defineTitle horizontal">
                  Events
                </dt>
                <dd
                  className="defineItem horizontal">
                  <a
                    className="textlink"
                    href="https://www.meetup.com/Numenta-Platform-for-Intelligent-Computing-Events/"
                    onClick={[Function]}
                    target={undefined}>
                    Meetup
                  </a>
                </dd>
                <dt
                  className="defineTitle horizontal">
                  Blog
                </dt>
                <dd
                  className="defineItem horizontal">
                  <a
                    className="textlink"
                    href={undefined}
                    onClick={[Function]}
                    style={Object {}}
                    target={undefined}>
                    Blog
                  </a>
                </dd>
                <dt
                  className="defineTitle horizontal">
                  HTM School
                </dt>
                <dd
                  className="defineItem horizontal">
                  <p
                    className="paragraph">
                    See the latest episode below!
                  </p>
                  <div
                    className="noLinkIcon">
                    <a
                      className="imagelink"
                      href="https://www.youtube.com/watch?v=HTW2Q_UrkAw"
                      onClick={[Function]}
                      title={undefined}>
                      <img
                        alt="Latest HTM School Episode"
                        className="image border mw shadow"
                        onClick={undefined}
                        src="https://i3.ytimg.com/vi/HTW2Q_UrkAw/mqdefault.jpg"
                        title={undefined} />
                    </a>
                  </div>
                </dd>
              </dl>
            </aside>
          </div>
          <div
            className="content">
            <p
              className="paragraph lead">
              Welcome to 
              <strong
                className="strong">
                Numenta.org
              </strong>
              , home of Numenta’s HTM community and open source projects. If you want to learn about Numenta the company visit 
               
              <a
                className="textlink"
                href="http://numenta.com/"
                onClick={[Function]}
                target={undefined}>
                Numenta.com
              </a>
               
               
              .
            </p>
            <h2
              className="h2">
              Machine Intelligence Starts Here
            </h2>
            <p
              className="paragraph">
              <strong
                className="strong">
                Hierarchical Temporal Memory
              </strong>
               is a foundational technology for the future of machine intelligence based upon the biology of the neocortex. Because Numenta is committed to making this technology accessible to everyone, all HTM software and ongoing research is open source. This allows you to work with our technology in whatever way works best for you ‒ learn about the theory, dive into the source code, or start your own implementation. Some of our community members have written their own versions of HTM systems in other languages and platforms. Others have created detailed visualizations, experiments, and applications.
            </p>
            <p
              className="paragraph">
              The neocortex is a logical system that we\'ll understand fully in time. HTM theory reflects our current understanding of how the neocortex works, and HTM code reduces that theory to practice. HTM is continually being updated as we learn more about the brain. We believe HTM will play a critical role in the creation of truly intelligent machines.
            </p>
            <h2
              className="h2">
              Join us!
            </h2>
            <p
              className="paragraph">
              There are many ways for you to get involved. Come discuss HTM theory with us 
               
              <a
                className="textlink"
                href="https://discourse.numenta.org/categories"
                onClick={[Function]}
                target={undefined}>
                on our forums
              </a>
               
               
              , or share your HTM application on our 
               
              <a
                className="textlink"
                href="https://github.com/numenta"
                onClick={[Function]}
                target={undefined}>
                Github organization
              </a>
               
               
              . If you\'re new to HTM, a great place to learn is in 
               
              <a
                className="textlink"
                href={undefined}
                onClick={[Function]}
                style={Object {}}
                target={undefined}>
                HTM School
              </a>
               
               
              . We hope you’ll join our community and help us discover the future of machine intelligence.
            </p>
          </div>
        </article>
      </div>
    </section>
    <section
      className="section"
      id="sectionHtm">
      <nav
        className="title"
        onClick={[Function]}
        title="Open Section: Hierarchical Temporal Memory">
        <svg
          className="toggle"
          fill="currentColor"
          height="1em"
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "color": undefined,
              "verticalAlign": "middle",
            }
          }
          viewBox="0 0 40 40"
          width="1em">
          <g>
            <path
              d="m31 16.4q0 0.3-0.2 0.5l-10.4 10.4q-0.3 0.3-0.5 0.3t-0.6-0.3l-10.4-10.4q-0.2-0.2-0.2-0.5t0.2-0.5l1.2-1.1q0.2-0.2 0.5-0.2t0.5 0.2l8.8 8.8 8.7-8.8q0.3-0.2 0.5-0.2t0.6 0.2l1.1 1.1q0.2 0.2 0.2 0.5z" />
          </g>
        </svg>
        <h1
          className="title">
          Hierarchical Temporal Memory
        </h1>
      </nav>
      <div
        className="article closed">
        <article>
          <div
            className="columns">
            <div
              className="aside">
              <img
                alt="HTM Image"
                className="image border mw shadow"
                onClick={undefined}
                src="filename.ext"
                title={undefined} />
            </div>
            <div
              className="content">
              <p
                className="paragraph lead">
                Hierarchical Temporal Memory (HTM) is a biologically-constrained theory of intelligence originally described in the book 
                 
                <a
                  className="textlink"
                  href="http://numenta.com/papers-videos-and-more/resources/on-intelligence/"
                  onClick={[Function]}
                  target={undefined}>
                  On Intelligence
                </a>
                . HTM is not a Deep Learning or Machine Learning technology. It is a machine intelligence framework strictly based on neuroscience and the physiology and interaction of pyramidal neurons in the neocortex of the mammalian brain.
              </p>
              <p
                className="paragraph">
                If you’re interested in learning more about HTM, visit our educational series 
                 
                <a
                  className="textlink"
                  href={undefined}
                  onClick={[Function]}
                  style={Object {}}
                  target={undefined}>
                  HTM School
                </a>
                 
                 
                or browse through the topics on 
                 
                <a
                  className="textlink"
                  href="https://discourse.numenta.org/categories"
                  onClick={[Function]}
                  target={undefined}>
                  HTM Forum
                </a>
                . You can also find a collection of research papers at 
                 
                <a
                  className="textlink"
                  href="http://numenta.com/papers/"
                  onClick={[Function]}
                  target={undefined}>
                  numenta.com/papers
                </a>
                 
                 
                and a living book authored by Numenta researchers and engineers that documents HTM called 
                 
                <a
                  className="textlink"
                  href="http://numenta.com/biological-and-machine-intelligence/"
                  onClick={[Function]}
                  target={undefined}>
                  Biological and Machine Intelligence (BAMI)
                </a>
                .
              </p>
            </div>
          </div>
        </article>
        <div
          className="closeButton"
          onClick={[Function]}
          title="Close Section: Hierarchical Temporal Memory">
          <svg
            className="closeIcon"
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": "inherit",
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 40 40"
            width="1em">
            <g>
              <path
                d="m31 26.4q0 0.3-0.2 0.5l-1.1 1.2q-0.3 0.2-0.6 0.2t-0.5-0.2l-8.7-8.8-8.8 8.8q-0.2 0.2-0.5 0.2t-0.5-0.2l-1.2-1.2q-0.2-0.2-0.2-0.5t0.2-0.5l10.4-10.4q0.3-0.2 0.6-0.2t0.5 0.2l10.4 10.4q0.2 0.2 0.2 0.5z" />
            </g>
          </svg>
          Close Section
        </div>
      </div>
    </section>
    <section
      className="section"
      id="sectionCommunity">
      <nav
        className="title open"
        onClick={[Function]}
        title="Close Section: Community">
        <svg
          className="toggle"
          fill="currentColor"
          height="1em"
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "color": undefined,
              "verticalAlign": "middle",
            }
          }
          viewBox="0 0 40 40"
          width="1em">
          <g>
            <path
              d="m31 26.4q0 0.3-0.2 0.5l-1.1 1.2q-0.3 0.2-0.6 0.2t-0.5-0.2l-8.7-8.8-8.8 8.8q-0.2 0.2-0.5 0.2t-0.5-0.2l-1.2-1.2q-0.2-0.2-0.2-0.5t0.2-0.5l10.4-10.4q0.3-0.2 0.6-0.2t0.5 0.2l10.4 10.4q0.2 0.2 0.2 0.5z" />
          </g>
        </svg>
        <h1
          className="title">
          Community
        </h1>
      </nav>
      <div
        className="article">
        <article>
          <div
            className="columns">
            <div
              className="aside">
              <img
                alt="HTM Community Event"
                className="image border mw shadow"
                onClick={undefined}
                src="filename.ext"
                title={undefined} />
            </div>
            <div
              className="content">
              <p
                className="paragraph lead">
                Our community is an eclectic collection of researchers, scientists, hobbyists, and hackers interested in building biologically-inspired intelligent systems with Hierarchical Temporal Memory (HTM).
              </p>
              <p
                className="paragraph">
                Sponsored by
                <a
                  className="textlink"
                  href="http://numenta.com/"
                  onClick={[Function]}
                  target={undefined}>
                   
                   
                  Numenta
                </a>
                 
                 
                and supported by a full-time employee dedicated to fostering and growing the community, we strive to create an inclusive environment. Our community exemplifies the type of people drawn toward new technology, and it shows in our curiosity and open-mindedness.
              </p>
              <p
                className="paragraph">
                To get involved in our community, join 
                 
                <a
                  className="textlink"
                  href="https://discourse.numenta.org/categories"
                  onClick={[Function]}
                  target={undefined}>
                  HTM Forum.
                </a>
                 
                 
                You can login with your Google, Facebook, or Twitter account, or by creating a new account with your email address. This is the best place to ask questions, search for answers, or just interact with others working on similar problems. We aim to be a helpful and welcoming community, and we hope you\'ll join us.
              </p>
            </div>
          </div>
          <span
            className="anchor"
            id="events" />
          <h2
            className="h2">
            Meetups, Hackathons, & Other Live Events
          </h2>
          <div
            className="columns">
            <div
              className="aside">
              <img
                alt="HTM Community Event #2"
                className="image border mw shadow"
                onClick={undefined}
                src="filename.ext"
                title={undefined} />
            </div>
            <div
              className="content">
              <p
                className="paragraph">
                While many of our interactions are virtual, we do hold community events like meetups, hackathons, and workshops throughout the year. All our live events are scheduled via Meetup. Please join our 
                 
                <a
                  className="textlink"
                  href="https://www.meetup.com/Numenta-Platform-for-Intelligent-Computing-Events/"
                  onClick={[Function]}
                  target={undefined}>
                  Meetup Group
                </a>
                 
                 
                for notifications of upcoming events, all of which are free and open to anyone. Please read our 
                 
                <a
                  className="textlink"
                  href={undefined}
                  onClick={[Function]}
                  style={Object {}}
                  target={undefined}>
                  code of conduct
                </a>
                 
                 
                before attending events.
              </p>
              <p
                className="paragraph">
                For recordings of previous events, please see our 
                 
                <a
                  className="textlink"
                  href="https://www.youtube.com/c/NumentaOrg"
                  onClick={[Function]}
                  target={undefined}>
                  YouTube channel.
                </a>
                 
                 
                You may also be interested in Numenta\'s 
                 
                <a
                  className="textlink"
                  href="http://numenta.com/events/"
                  onClick={[Function]}
                  target={undefined}>
                  company events and archive.
                </a>
              </p>
            </div>
          </div>
        </article>
        <div
          className="closeButton"
          onClick={[Function]}
          title="Close Section: Community">
          <svg
            className="closeIcon"
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": "inherit",
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 40 40"
            width="1em">
            <g>
              <path
                d="m31 26.4q0 0.3-0.2 0.5l-1.1 1.2q-0.3 0.2-0.6 0.2t-0.5-0.2l-8.7-8.8-8.8 8.8q-0.2 0.2-0.5 0.2t-0.5-0.2l-1.2-1.2q-0.2-0.2-0.2-0.5t0.2-0.5l10.4-10.4q0.3-0.2 0.6-0.2t0.5 0.2l10.4 10.4q0.2 0.2 0.2 0.5z" />
            </g>
          </svg>
          Close Section
        </div>
      </div>
    </section>
    <section
      className="section"
      id="sectionCode">
      <nav
        className="title"
        onClick={[Function]}
        title="Open Section: Code">
        <svg
          className="toggle"
          fill="currentColor"
          height="1em"
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "color": undefined,
              "verticalAlign": "middle",
            }
          }
          viewBox="0 0 40 40"
          width="1em">
          <g>
            <path
              d="m31 16.4q0 0.3-0.2 0.5l-10.4 10.4q-0.3 0.3-0.5 0.3t-0.6-0.3l-10.4-10.4q-0.2-0.2-0.2-0.5t0.2-0.5l1.2-1.1q0.2-0.2 0.5-0.2t0.5 0.2l8.8 8.8 8.7-8.8q0.3-0.2 0.5-0.2t0.6 0.2l1.1 1.1q0.2 0.2 0.2 0.5z" />
          </g>
        </svg>
        <h1
          className="title">
          Code
        </h1>
      </nav>
      <div
        className="article closed">
        <article>
          <div
            className="columns">
            <div
              className="aside">
              <img
                alt="HTM Coding"
                className="image border mw shadow"
                onClick={undefined}
                src="filename.ext"
                title={undefined} />
            </div>
            <div
              className="content">
              <p
                className="paragraph lead">
                In addition to making all of Numenta\'s 
                 
                <a
                  className="textlink"
                  href="https://github.com/numenta"
                  onClick={[Function]}
                  target={undefined}>
                  production
                </a>
                 
                 
                and 
                 
                <a
                  className="textlink"
                  href="https://github.com/numenta/nupic.research"
                  onClick={[Function]}
                  target={undefined}>
                  research
                </a>
                 
                 
                code open source, our community produces and shares HTM implementations and applications, commonly posting their projects on 
                 
                <a
                  className="textlink"
                  href="https://discourse.numenta.org/categories"
                  onClick={[Function]}
                  target={undefined}>
                  HTM Forum.
                </a>
                 
                 
                Our community hosts many of these projects on the 
                 
                <a
                  className="textlink"
                  href="https://github.com/htm-community"
                  onClick={[Function]}
                  target={undefined}>
                  HTM Community
                </a>
                 
                 
                Github organization. We have 
                 
                <a
                  className="textlink"
                  href={undefined}
                  onClick={[Function]}
                  style={Object {}}
                  target={undefined}>
                  HTM implementations
                </a>
                 
                 
                in languages like Python, C++, Java, Clojure, Go, and JavaScript.
              </p>
              <p
                className="paragraph">
                We are always looking for community members that can help others build out HTM applications and experiments. If you are interested in helping or providing consulting services, please leave us a note on the 
                 
                <a
                  className="textlink"
                  href="https://discourse.numenta.org/c/htm-hackers"
                  onClick={[Function]}
                  target={undefined}>
                  HTM Hackers Forum
                </a>
                 
                 
                and introduce yourself.
              </p>
            </div>
          </div>
        </article>
        <div
          className="closeButton"
          onClick={[Function]}
          title="Close Section: Code">
          <svg
            className="closeIcon"
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": "inherit",
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 40 40"
            width="1em">
            <g>
              <path
                d="m31 26.4q0 0.3-0.2 0.5l-1.1 1.2q-0.3 0.2-0.6 0.2t-0.5-0.2l-8.7-8.8-8.8 8.8q-0.2 0.2-0.5 0.2t-0.5-0.2l-1.2-1.2q-0.2-0.2-0.2-0.5t0.2-0.5l10.4-10.4q0.3-0.2 0.6-0.2t0.5 0.2l10.4 10.4q0.2 0.2 0.2 0.5z" />
            </g>
          </svg>
          Close Section
        </div>
      </div>
    </section>
  </div>
</div>
`;
